toolchain/check/import_ref.cpp

```diff
@@ -1811,6 +1798,106 @@ class ImportRefResolver {
+    auto import_block_id = import_generic.GetEvalBlock(region);
+    if (!import_block_id.is_valid()) {
+      return SemIR::InstBlockId::Invalid;
+    }
+
+    auto inst_ids = ResolveLocalInstBlockContents(import_block_id);
+    return RebuildGenericEvalBlock(context_, generic_id, region, inst_ids);
+  }
+
+  // Fills in the remaining information in a partially-imported generic.
+  auto FinishPendingGeneric(PendingGeneric pending) -> void {
+    const auto& import_generic = import_ir_.generics().Get(pending.import_id);
+
+    // Note that the generics list can be reallocated by ResolveLocalEvalBlock
+    // importing more generics.
+    auto decl_block_id =
+        ResolveLocalEvalBlock(import_generic, pending.local_id,
+                              SemIR::GenericInstIndex::Region::Declaration);
+    auto* local_generic = &context_.generics().Get(pending.local_id);
+    local_generic->decl_block_id = decl_block_id;
+
+    auto self_specific_id = MakeSelfSpecific(context_, pending.local_id);
+    local_generic->self_specific_id = self_specific_id;
+    pending_specifics_.push_back({.import_id = import_generic.self_specific_id,
+                                  .local_id = self_specific_id});
+
+    auto definition_block_id =
+        ResolveLocalEvalBlock(import_generic, pending.local_id,
+                              SemIR::GenericInstIndex::Region::Definition);
+    local_generic = &context_.generics().Get(pending.local_id);
+    local_generic->definition_block_id = definition_block_id;
+  }
+
+  // Resolves and returns a local inst block of constant instructions
+  // corresponding to an imported inst block.
+  auto ResolveLocalInstBlock(SemIR::InstBlockId import_block_id)
+      -> SemIR::InstBlockId {
+    if (!import_block_id.is_valid()) {
+      return SemIR::InstBlockId::Invalid;
+    }
+
+    auto inst_ids = ResolveLocalInstBlockContents(import_block_id);
+    return context_.inst_blocks().Add(inst_ids);
+  }
+
+  // Fills in the remaining information in a partially-imported specific.
+  auto FinishPendingSpecific(PendingSpecific pending) -> void {
+    const auto& import_specific = import_ir_.specifics().Get(pending.import_id);
+
+    // Note that the specifics list can be reallocated by ResolveLocalInstBlock
+    // importing more specifics.
+    auto* local_specific = &context_.specifics().Get(pending.local_id);
+    if (!local_specific->decl_block_id.is_valid()) {
+      auto decl_block_id = ResolveLocalInstBlock(import_specific.decl_block_id);
+      local_specific = &context_.specifics().Get(pending.local_id);
+      local_specific->decl_block_id = decl_block_id;
+    }
+
+    if (!local_specific->definition_block_id.is_valid() &&
+        import_specific.definition_block_id.is_valid()) {
+      auto definition_block_id =
+          ResolveLocalInstBlock(import_specific.definition_block_id);
+      local_specific = &context_.specifics().Get(pending.local_id);
+      local_specific->definition_block_id = definition_block_id;
```

Comment:
Similar to the above, I'd suggest only reusing the same variable when it's significant (which for line 1892, it is, but that doesn't look true here).

```suggestion
      context_.specifics().Get(pending.local_id).definition_block_id = definition_block_id;
```

Reply:
Note, suggesting this mainly because I find it easier to figure out what the source is when it's assigned in fewer places.